ament_cmake_auto/cmake/ament_auto_find_build_dependencies.cmake

```diff
@@ -27,18 +32,35 @@
 # @public
 #
 macro(ament_auto_find_build_dependencies)
-  if(ARGN)
+  cmake_parse_arguments(ARG "" "" "REQUIRED_PACKAGES" ${ARGN})
+  if(ARG_UNPARSED_ARGUMENTS)
     message(FATAL_ERROR "ament_auto_find_build_dependencies() called with "
-      "unused arguments: ${ARGN}")
+      "unused arguments: ${ARG_UNPARSED_ARGUMENTS}")
   endif()
 
   if(NOT _AMENT_PACKAGE_NAME)
     ament_package_xml()
   endif()
 
+  # ensure that the caller isn't expecting additional packages to be found
+  set(_build_and_buildtool_depends
+    ${${PROJECT_NAME}_BUILD_DEPENDS}
+    ${${PROJECT_NAME}_BUILDTOOL_DEPENDS})
+  foreach(_package_name ${ARG_REQUIRED_PACKAGES})
+    if(NOT _package_name IN_LIST _build_and_buildtool_depends)
```

Comment:
I know that [this use of IN_LIST](https://github.com/ament/ament_cmake/blob/40230914a02d209e2927a1b70e8be93d50db24b2/ament_cmake_core/cmake/core/list_append_unique.cmake#L30) puts the variable in brackets, but if I did that here it didn't work as expected. There are subtleties here that I am not completely comfortable with so please double-check that this is reasonable